examples/formbot/actions/actions.py

```diff
@@ -1,140 +1,117 @@
-                self.from_entity(entity="seating"),
-                self.from_intent(intent="affirm", value=True),
-                self.from_intent(intent="deny", value=False),
-            ],
-            "preferences": [
-                self.from_intent(intent="deny", value="no additional preferences"),
-                self.from_text(not_intent="affirm"),
-            ],
-            "feedback": [self.from_entity(entity="feedback"), self.from_text()],
-        }
-
-    @staticmethod
-    def cuisine_db() -> List[Text]:
-        """Database of supported cuisines."""
-
-        return [
-            "caribbean",
-            "chinese",
-            "french",
-            "greek",
-            "indian",
-            "italian",
-            "mexican",
-        ]
-
-    @staticmethod
-    def is_int(string: Text) -> bool:
-        """Check if a string is an integer."""
-
-        try:
-            int(string)
-            return True
-        except ValueError:
-            return False
-
-    def validate_cuisine(
-        self,
-        value: Text,
-        dispatcher: CollectingDispatcher,
-        tracker: Tracker,
-        domain: Dict[Text, Any],
-    ) -> Dict[Text, Any]:
-        """Validate cuisine value."""
-
-        if value.lower() in self.cuisine_db():
-            # validation succeeded, set the value of the "cuisine" slot to value
-            return {"cuisine": value}
-        else:
-            dispatcher.utter_message(template="utter_wrong_cuisine")
-            # validation failed, set this slot to None, meaning the
-            # user will be asked for the slot again
-            return {"cuisine": None}
-
-    def validate_num_people(
-        self,
-        value: Text,
-        dispatcher: CollectingDispatcher,
-        tracker: Tracker,
-        domain: Dict[Text, Any],
-    ) -> Dict[Text, Any]:
-        """Validate num_people value."""
-
-        if self.is_int(value) and int(value) > 0:
-            return {"num_people": value}
-        else:
-            dispatcher.utter_message(template="utter_wrong_num_people")
-            # validation failed, set slot to None
-            return {"num_people": None}
-
-    def validate_outdoor_seating(
-        self,
-        value: Text,
-        dispatcher: CollectingDispatcher,
-        tracker: Tracker,
-        domain: Dict[Text, Any],
-    ) -> Dict[Text, Any]:
-        """Validate outdoor_seating value."""
-
-        if isinstance(value, str):
-            if "out" in value:
-                # convert "out..." to True
-                return {"outdoor_seating": True}
-            elif "in" in value:
-                # convert "in..." to False
-                return {"outdoor_seating": False}
-            else:
-                dispatcher.utter_message(template="utter_wrong_outdoor_seating")
-                # validation failed, set slot to None
-                return {"outdoor_seating": None}
+        return "action_submit_restaurant_form"
```

Comment:
I am not sure that we should still use the term `form` in the name of the class and the comments. It is just a regular action, and we need a rule to execute it.

By using the name `form` in the action code, it is possible to think that this is some kind of special Action.

Another approach would be to just call this: `ActionRestaurant`

And then have a `ValidateRestaurantForm`. It does make sense to use the therm `Form` in the Validate action, because it is called automatically, so it is a special kind of action.